src/core/gps/qgsgpslogger.cpp

```diff
@@ -0,0 +1,367 @@
+{
+  mDistanceCalculator.setEllipsoid( ellipsoid );
+  emit distanceAreaChanged();
+}
+
+void QgsGpsLogger::setTransformContext( const QgsCoordinateTransformContext &context )
+{
+  mTransformContext = context;
+  mDistanceCalculator.setSourceCrs( mWgs84CRS, mTransformContext );
+  emit distanceAreaChanged();
+}
+
+QgsCoordinateTransformContext QgsGpsLogger::transformContext() const
+{
+  return mTransformContext;
+}
+
+const QgsDistanceArea &QgsGpsLogger::distanceArea() const
+{
+  return mDistanceCalculator;
+}
+
+QVector<QgsPoint> QgsGpsLogger::currentTrack() const
+{
+  return mCaptureListWgs84;
+}
+
+QgsPointXY QgsGpsLogger::lastPosition() const
+{
+  return mLastGpsPositionWgs84;
+}
+
+double QgsGpsLogger::lastElevation() const
+{
+  return mLastElevation;
+}
+
+void QgsGpsLogger::resetTrack()
+{
+  mBlockGpsStateChanged++;
+
+  const bool trackWasEmpty = mCaptureListWgs84.isEmpty();
+  mCaptureListWgs84.clear();
+  mBlockGpsStateChanged--;
+  mTrackStartTime = QDateTime();
+
+  if ( !trackWasEmpty )
+    emit trackIsEmptyChanged( true );
+
+  emit trackReset();
+}
+
+void QgsGpsLogger::updateGpsSettings()
+{
+  int acquisitionInterval = 0;
+  if ( QgsGpsConnection::settingsGpsTimeStampSpecification.exists() )
+  {
+    acquisitionInterval = static_cast< int >( QgsGpsConnection::settingGpsAcquisitionInterval.value() );
+    mDistanceThreshold = QgsGpsConnection::settingGpsDistanceThreshold.value();
+    mApplyLeapSettings = QgsGpsConnection::settingGpsApplyLeapSecondsCorrection.value();
+    mLeapSeconds = static_cast< int >( QgsGpsConnection::settingGpsLeapSeconds.value() );
+    mTimeStampSpec = QgsGpsConnection::settingsGpsTimeStampSpecification.value();
+    mTimeZone = QgsGpsConnection::settingsGpsTimeStampTimeZone.value();
+    mOffsetFromUtc = static_cast< int >( QgsGpsConnection::settingsGpsTimeStampOffsetFromUtc.value() );
+  }
+  else
+  {
+    // legacy settings
+    QgsSettings settings;
+
+    acquisitionInterval = settings.value( QStringLiteral( "acquisitionInterval" ), 0, QgsSettings::Gps ).toInt();
+    mDistanceThreshold = settings.value( QStringLiteral( "distanceThreshold" ), 0, QgsSettings::Gps ).toDouble();
+    mApplyLeapSettings = settings.value( QStringLiteral( "applyLeapSeconds" ), true, QgsSettings::Gps ).toBool();
+    mLeapSeconds = settings.value( QStringLiteral( "leapSecondsCorrection" ), 18, QgsSettings::Gps ).toInt();
+
+    switch ( settings.value( QStringLiteral( "timeStampFormat" ), Qt::LocalTime, QgsSettings::Gps ).toInt() )
+    {
+      case 0:
+        mTimeStampSpec = Qt::TimeSpec::LocalTime;
+        break;
+
+      case 1:
+        mTimeStampSpec = Qt::TimeSpec::UTC;
+        break;
+
+      case 2:
+        mTimeStampSpec = Qt::TimeSpec::TimeZone;
+        break;
+    }
+    mTimeZone = settings.value( QStringLiteral( "timestampTimeZone" ), QVariant(), QgsSettings::Gps ).toString();
+  }
+
+  mAcquisitionInterval = acquisitionInterval * 1000;
+  if ( mAcquisitionTimer->isActive() )
```

Comment:
Thought: we should move away from this QTimer thing, we have the gpsStateChanged signal that's already driving things.

Reply:
Agreed - I'm hoping this refactor will open us up for more cleanup of this old code (and better test coverage to accompany it!). (The distance based filtering also should be updated to use the QgsDistanceArea calculator too)